neqo-http3/src/connection_client.rs

```diff
@@ -296,10 +296,10 @@ impl Http3Client {
 
         // Transform pseudo-header fields
         let mut final_headers = vec![
-            (":method".into(), method.to_owned()),
-            (":scheme".into(), scheme.to_owned()),
-            (":authority".into(), host.to_owned()),
-            (":path".into(), path.to_owned()),
+            Header(":method".into(), method.to_owned()),
+            Header(":scheme".into(), scheme.to_owned()),
+            Header(":authority".into(), host.to_owned()),
+            Header(":path".into(), path.to_owned()),
```

Comment:
You might do better with `Header::new(impl ToString, impl ToString)` here so that you can avoid all the .into() or .to_owned() calls.  That will create new instances as needed (or just convert an already owned value without copying).

You wouldn't have to change all those call sites, but you could.